tools/pmd_test.bzl

```diff
@@ -7,38 +7,71 @@ def _impl(ctx):
         fail("Expecting a single java library")
 
     src_jar = lib.source_jars[0]
-    full_compile_jars = ":".join([f.path for f in lib.full_compile_jars.to_list()])
+    jar_deps = lib.full_compile_jars
+    full_compile_jars = ":".join([f.short_path for f in jar_deps.to_list()])
 
     # Presumably this needs to be updated to actually run the resulting shell script.
-    cmd = "@//tools:pmd"
 
-    args = [
-        cmd,
+    pmd_exe_file = ctx.attr._pmd[DefaultInfo].files_to_run.executable
+    ruleset = ctx.file.ruleset
+    pmd_cmd_args = [
+        pmd_exe_file.short_path,
         "-f text",
-        "-R pmd-ruleset.xml",
+        "-R {}".format(ruleset.short_path),
         "-d {}".format(src_jar.short_path),
-	"-auxclasspath {}".format(full_compile_jars),
+        "-auxclasspath {}".format(full_compile_jars),
+    ]
+    pmd_cmd = " ".join(pmd_cmd_args)
+    script = [
+        "#!/usr/bin/env bash",
+        "set -o errexit",
+        "set -o xtrace",
+        pmd_cmd,
     ]
-    argline = " ".join(args)
 
-    script = '\n'.join([
-        argline,
-    ])
-    print(script)
+    script_content = "\n".join(script)
+    print(script_content)
 
     # Write the file, it is executed by 'bazel test'.
+    test_file = ctx.actions.declare_file(ctx.attr.name)
     ctx.actions.write(
-        output = ctx.outputs.executable,
-        content = script,
+        output = test_file,
+        content = script_content,
+        is_executable = True,
```

Comment:
(Presumably the `is_executable` is on by default if you use `ctx.output.executable` instead.)